scraper/src/youtube2zim/scraper.py

```diff
@@ -1108,3 +1120,240 @@ def to_data_js(video):
+
+        def get_thumbnail_path(video_id):
+            return f"videos/{video_id}/video.webp"
+
+        def get_subtitles(video_id):
+            video_dir = self.videos_dir.joinpath(video_id)
+            languages = [
+                x.stem.split(".")[1]
+                for x in video_dir.iterdir()
+                if x.is_file() and x.name.endswith(".vtt")
+            ]
+
+            def to_subtitle_object(lang):
+                try:
+                    try:
+                        subtitle = get_language_details(
+                            YOUTUBE_LANG_MAP.get(lang, lang)
+                        )
+                    except NotFound:
+                        lang_simpl = re.sub(r"^([a-z]{2})-.+$", r"\1", lang)
+                        subtitle = get_language_details(
+                            YOUTUBE_LANG_MAP.get(lang_simpl, lang_simpl)
+                        )
+                except Exception:
+                    logger.error(f"Failed to get language details for {lang}")
+                    raise
+                return Subtitle(
+                    code=lang,
+                    name=f"{subtitle['english'].title()} - {subtitle['query']}",
+                )
+
+            # Youtube.com sorts subtitles by English name
+            return sorted(map(to_subtitle_object, languages), key=lambda x: x.name)
+
+        def get_videos_list(playlist):
+            videos = load_mandatory_json(
+                self.cache_dir, f"playlist_{playlist.playlist_id}_videos"
+            )
+            videos = list(filter(skip_deleted_videos, videos))
+            videos = list(filter(is_present, videos))
+            videos = list(filter(has_channel, videos))
+            videos = sorted(videos, key=lambda v: v["snippet"]["position"])
+            return videos
+
+        def generate_video_object(video):
+            video_id = video["contentDetails"]["videoId"]
+            title = video["snippet"]["title"]
+            description = video["snippet"]["description"]
+            publication_date = video["contentDetails"]["videoPublishedAt"]
+            author = videos_channels[video_id]
+
+            return Video(
+                id=video_id,
+                title=title,
+                description=description,
+                author=Author(
+                    channel_id=author["channelId"],
+                    channel_title=author["channelTitle"],
+                    profile_path=f"channels/{author['channelId']}/profile.jpg",
+                    banner_path=f"channels/{author['channelId']}/banner.jpg",
+                ),
+                publication_date=publication_date,
+                video_path=f"videos/{video_id}/video.{self.video_format}",
+                thumbnail_path=get_thumbnail_path(video_id),
+                subtitle_path=f"videos/{video_id}",
+                subtitle_list=get_subtitles(video_id),
+                duration=0,
+            )
+
+        def generate_video_preview_object(video):
```

Comment:
Type hint for returned value again

Reply:
Added in [e62b0bb](https://github.com/openzim/youtube/pull/219/commits/e62b0bb2b30b069f1377f4ce296ee69f2852f833)